src/profile-logic/import/simpleperf.js

```diff
@@ -0,0 +1,644 @@
+// @flow
+
+import { simpleperf_report_proto as report } from './proto/simpleperf_report';
+
+import type { Milliseconds } from 'firefox-profiler/types/units';
+import type {
+  CategoryList,
+  FrameTable,
+  FuncTable,
+  IndexIntoCategoryList,
+  IndexIntoFrameTable,
+  IndexIntoFuncTable,
+  IndexIntoResourceTable,
+  IndexIntoStackTable,
+  ProfileMeta,
+  ResourceTable,
+  SerializableSamplesTable,
+  SerializableProfile,
+  SerializableThread,
+  StackTable,
+} from 'firefox-profiler/types/profile';
+import { UniqueStringArray } from 'firefox-profiler/utils/unique-string-array';
+
+import Long from 'long';
+
+const $NotNull = <T>(val: T): $NonMaybeType<T> => val;
+
+function toNumber(value: Long | number): number {
+  if (Long.isLong(value)) {
+    const longValue: Long = value;
+    return longValue.toNumber();
+  }
+
+  return value;
+}
+
+function toMilliseconds(nanoseconds: number | Long): Milliseconds {
+  return toNumber(nanoseconds) / 1000_000;
+}
+
+class Categories {
+  static categoryList: CategoryList = [];
+
+  static Other = this.createCategory('Other', 'grey');
+  static Native = this.createCategory('Native', 'magenta');
+  static Java = this.createCategory('Java', 'green');
+  static System = this.createCategory('System', 'yellow');
+  static Kernel = this.createCategory('Kernel', 'orange');
+
+  static toJson(): CategoryList {
+    return this.categoryList;
+  }
+
+  static createCategory(name: string, color: string): IndexIntoCategoryList {
+    const index = this.categoryList.length;
+    this.categoryList.push({ name, color, subcategories: ['Other'] });
+
+    return index;
+  }
+}
+
+class FirefoxResourceTable {
+  strings: UniqueStringArray;
+
+  resourceTable: ResourceTable = {
+    length: 0,
+    lib: [],
+    name: [],
+    host: [],
+    type: [],
+  };
+  resourcesMap: Map<number, IndexIntoResourceTable> = new Map();
+
+  constructor(strings: UniqueStringArray) {
+    this.strings = strings;
+  }
+
+  toJson(): ResourceTable {
+    return this.resourceTable;
+  }
+
+  findOrAddResource(file: report.IFile): IndexIntoResourceTable {
+    if (!this.resourcesMap.has(file.id)) {
```

Comment:
If you do something such as :
```js
let resourceIndex = this.resourcesMap.get(file.id);
if (!resourceIndex) {
  ...
  resourceIndex = this.resourceTable.length++;
  this.resourcesMap.set(file.id, resourceIndex);   
}
return resourceIndex;
```
then you don't need your `$NotNull` utility, with the added bonus that the hashmap is requested only once.

Same comment for all the code in this file that applies.